src/Http/Message/AbstractRequest.php

```diff
@@ -0,0 +1,265 @@
+     *
+     * @see http://tools.ietf.org/html/rfc3986#section-4.3
+     *
+     * @var UriInterface
+     */
+    protected UriInterface $uri;
+
+    /**
+     * @return string
+     */
+    public function getMethod(): string
+    {
+        return $this->method;
+    }
+
+    /**
+     * Retrieves the message's request target.
+     *
+     * Retrieves the message's request-target either as it will appear (for
+     * clients), as it appeared at request (for servers), or as it was
+     * specified for the instance (see withRequestTarget()).
+     *
+     * In most cases, this will be the origin-form of the composed URI, unless a
+     * value was provided to the concrete implementation (see
+     * withRequestTarget() below).
+     *
+     * @return string
+     */
+    public function getRequestTarget(): string
+    {
+        $requestTarget = $this->requestTarget;
+
+        if (null === $requestTarget) {
+            $requestTarget = $this->uri->getPath();
+
+            if (true !== empty($this->uri->getQuery())) {
+                $requestTarget .= "?" . $this->uri->getQuery();
+            }
+
+            if (true === empty($requestTarget)) {
+                $requestTarget = "/";
+            }
+        }
+
+        return $requestTarget;
+    }
+
+    /**
+     * Return an instance with the provided HTTP method.
+     *
+     * While HTTP method names are typically all uppercase characters, HTTP
+     * method names are case-sensitive and thus implementations SHOULD NOT
+     * modify the given string.
+     *
+     * This method MUST be implemented in such a way as to retain the
+     * immutability of the message, and MUST return an instance that has the
+     * changed request method.
+     *
+     * @param string $method
+     *
+     * @return RequestInterface
+     * @throws InvalidArgumentException for invalid HTTP methods.
+     */
+    public function withMethod(string $method): RequestInterface
+    {
+        $this->processMethod($method);
+
+        return $this->cloneInstance($method, "method");
+    }
+
+    /**
+     * Return an instance with the specific request-target.
+     *
+     * If the request needs a non-origin-form request-target — e.g., for
+     * specifying an absolute-form, authority-form, or asterisk-form —
+     * this method may be used to create an instance with the specified
+     * request-target, verbatim.
+     *
+     * This method MUST be implemented in such a way as to retain the
+     * immutability of the message, and MUST return an instance that has the
+     * changed request target.
+     *
+     * @see http://tools.ietf.org/html/rfc7230#section-5.3 (for the various
+     *     request-target forms allowed in request messages)
+     *
+     * @param string|null $requestTarget
+     *
+     * @return RequestInterface
+     */
+    public function withRequestTarget(?string $requestTarget): RequestInterface
+    {
+        if (preg_match("/\s/", $requestTarget)) {
```

Comment:
In case of second param is null, it will generate error. Function accepts only string.

Reply:
Fixed